assembly/buffer/index.ts

```diff
@@ -11,6 +12,42 @@ export class Buffer extends Uint8Array {
     return new Buffer(size);
   }
 
+  public static concat<T>(items: T, length: i32): Buffer {
```

Comment:
Thinking of this a bit, I've only ever used this to concatenate two or multiple `Buffer`s in the past. Might, or might not, be something to watch out for when implementing those APIs, as limiting to `Buffer` makes this and other APIs quite a bit simpler.

Reply:
I was thinking along the lines of:

```ts
public static concat(items: Uint8Array[], length: i32): Buffer
```

I'm not sure what the compiler will say.

Reply:
Looks fine, yeah, do prefer that as well. If the compiler doesn't accept a `Buffer[]` here,, we should make it to.

Reply:
```
Test File : tests\buffer.spec.ts (untouched)
ERROR AS200: Conversion from type '~lib/array/Array<~lib/buffer/index/Buffer>' to '~lib/array/Array<~lib/typedarray/Uint8Array>' requires an explicit cast.

     let actual: Buffer = Buffer.concat(list, 15);
```

I'm curious if there's a way to get the compiler to accept this? TypeScript accepts this just fine. (and honestly it should be safe.)

Reply:
```
Test File : tests\buffer.spec.ts (untouched)
ERROR AS200: Conversion from type '~lib/array/Array<~lib/buffer/index/Buffer>' to '~lib/array/Array<~lib/typedarray/Uint8Array>' requires an explicit cast.

     let actual: Buffer = Buffer.concat(list, 15);
```

I'm curious if there's a way to get the compiler to accept this? TypeScript accepts this just fine. (and honestly it should be safe.)

In the meantime let's switch to `Buffer[]` and move forward. We can fix this later.